### DISCUSSION TITLE
[v4] Installing Tailwind v4 Beta with the CLI generates huge output file

### DISCUSSION THREAD

**Discussion by meteornu** [tailwindlabs/tailwindcss#15213] [Help] [answered]
I followed the steps for the install process found here:

**Installing the CLI**
https://tailwindcss.com/docs/v4-beta#installing-the-cli

- I start by running: `npm install tailwindcss@next @tailwindcss/cli@next`
- Then add `@import "tailwindcss";` in my `input.css` file (the guide specifies an `app.css` file, but I suppose they mean this is the input file?)
- Then finalize by running `npx @tailwindcss/cli -i input.css -o output.css --watch`

For testing purposes, my project is completely empty. No classes to be picked up by the watch, but Tailwind still adds all helper classes in my `output.css` file for some reason. The file grows to **68.2 kB** which it shouldn't.

I compare it to the file generated in the playground here:
https://play.tailwindcss.com/ARFBI7tBpA which is only **3.94 kB**.

What am I missing? Is the guide incomplete, or am I misunderstanding something? Let me know if there's details missing to help me further.

Appreciate the help!

**Comment by philipp-spiess** [comment]
Hey! Can you share some infos about your project? How does the folder structure look like? By default, Tailwind will scan all directories in the root directory of your project except those included in a `.gitignore`, so perhaps you have some kind of `dist` folder that is not ignored?

**Reply by meteornu** [reply]
Thanks for your answer! As I mentioned, my project is completely bare minimum, only what is required to set up Tailwind. This is my structure:

```
> root folder
	> tailwind
		> node_modules
		app.css
		package.json
		package-lock.json
	style.css
``` 
I run the command `npx @tailwindcss/cli -i app.css -o ../style.css --watch` inside the `tailwind`-folder. But once Tailwind is compiled, this is what it looks like: [style.txt](https://github.com/user-attachments/files/17944080/style.txt)

**Reply by philipp-spiess** [reply, marked as answer]
Do you have a `.gitignore` file too that includes the `node_modules` folder? Otherwise all your dependencies will also be scanned which might cause this unexpected thing!

You can customize what folder to scan for templates files as well from within your app.css file: https://tailwindcss-com-git-v4-beta-docs-tailwindlabs.vercel.app/docs/v4-beta#configuring-source-detection

**Reply by meteornu** [reply]
You are right. If I add a `.gitignore` file with the following line:
```
/node_modules
```
The file shrank to **21 kB**. Thanks. 😃
